config/graceful_shutdown_config.go

```diff
@@ -54,7 +58,7 @@ type ShutdownConfig struct {
 	RejectRequest bool
 	// true -> all requests had been processed. In provider side it means that all requests are returned response to clients
 	// In consumer side, it means that all requests getting response from servers
-	RequestsFinished bool
+	RequestsFinished *atomic.Bool
```

Comment:
RequestsFinished is pointer,  can't be fill with config, requires explicit assignment.

It's better to use value types.

Reply:
Yes, it is a bug. The latest commit will solve this problem.